services/QuillLMS/app/controllers/admin_diagnostic_reports_controller.rb

```diff
@@ -6,14 +6,24 @@ class AdminDiagnosticReportsController < ApplicationController
     "report" => AdminDiagnosticReports::DiagnosticOverviewWorker
   }
 
-  before_action :set_query
-  before_action :validate_request
-  before_action :authorize_request
+  BASE_REPORT_FILTER_NAME = 'diagnostic_growth_report'
+  OVERVIEW_REPORT_FILTER_NAME = 'diagnostic_growth_report_overview'
+  SKILL_REPORT_FILTER_NAME = 'diagnostic_growth_report_skill'
+  STUDENT_REPORT_FILTER_NAME = 'diagnostic_growth_report_student'
+
+  before_action :set_query, only: [:report]
+  before_action :validate_request, only: [:report]
+  before_action :authorize_request, only: [:report]
```

Comment:
Good question.  I think that in this case, because the code logic generates and sends a report to `current_user`, we don't need auth.  You can never get access to data that's not yours.  Open to discussion, though.